com.ibm.streamsx.topology/opt/python/packages/streamsx/rest_primitives.py

```diff
@@ -561,8 +561,10 @@ def retrieve_log_trace(self, filename=None, dir=None):
 
         if not filename:
             filename = _file_name('job', self.id, '.tar.gz')
- 
-        return self.rest_client._retrieve_file(self.applicationLogTrace, filename, dir, 'application/x-compressed')
+        if hasattr(self, "applicationLogTrace") and self.applicationLogTrace is not None:
```

Comment:
Ideally the creating a default `filename` would occur only if the method is going to retrieve logs.